File: clients/algoliasearch-client-php/lib/Model/Personalization/DeleteUserProfileResponse.php
<?php

namespace Algolia\AlgoliaSearch\Model\Personalization;

/**
 * DeleteUserProfileResponse Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class DeleteUserProfileResponse extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'userToken' => 'string',
        'deletedUntil' => 'string',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'userToken' => null,
        'deletedUntil' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'userToken' => 'setUserToken',
        'deletedUntil' => 'setDeletedUntil',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'userToken' => 'getUserToken',
        'deletedUntil' => 'getDeletedUntil',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['userToken'])) {
            $this->container['userToken'] = $data['userToken'];
        }
        if (isset($data['deletedUntil'])) {
            $this->container['deletedUntil'] = $data['deletedUntil'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['userToken']) || $this->container['userToken'] === null) {
            $invalidProperties[] = "'userToken' can't be null";
        }
        if (!isset($this->container['deletedUntil']) || $this->container['deletedUntil'] === null) {
            $invalidProperties[] = "'deletedUntil' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets userToken
     *
     * @return string
     */
    public function getUserToken()
    {
        return $this->container['userToken'] ?? null;
    }

    /**
     * Sets userToken
     *
     * @param string $userToken userToken representing the user for which to fetch the Personalization profile
     *
     * @return self
     */
    public function setUserToken($userToken)
    {
        $this->container['userToken'] = $userToken;

        return $this;
    }

    /**
     * Gets deletedUntil
     *
     * @return string
     */
    public function getDeletedUntil()
    {
        return $this->container['deletedUntil'] ?? null;
    }

    /**
     * Sets deletedUntil
     *
     * @param string $deletedUntil A date until which the data can safely be considered as deleted for the given user. Any data received after the deletedUntil date will start building a new user profile.
     *
     * @return self
     */
    public function setDeletedUntil($deletedUntil)
    {
        $this->container['deletedUntil'] = $deletedUntil;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/personalization/DeleteUserProfileResponse.java
package com.algolia.model.personalization;

import com.google.gson.annotations.SerializedName;
import java.util.Objects;

/** DeleteUserProfileResponse */
public class DeleteUserProfileResponse {

  @SerializedName("userToken")
  private String userToken;

  @SerializedName("deletedUntil")
  private String deletedUntil;

  public DeleteUserProfileResponse setUserToken(String userToken) {
    this.userToken = userToken;
    return this;
  }

  /**
   * userToken representing the user for which to fetch the Personalization profile.
   *
   * @return userToken
   */
  @javax.annotation.Nonnull
  public String getUserToken() {
    return userToken;
  }

  public DeleteUserProfileResponse setDeletedUntil(String deletedUntil) {
    this.deletedUntil = deletedUntil;
    return this;
  }

  /**
   * A date until which the data can safely be considered as deleted for the given user. Any data
   * received after the deletedUntil date will start building a new user profile.
   *
   * @return deletedUntil
   */
  @javax.annotation.Nonnull
  public String getDeletedUntil() {
    return deletedUntil;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DeleteUserProfileResponse deleteUserProfileResponse = (DeleteUserProfileResponse) o;
    return (
      Objects.equals(this.userToken, deleteUserProfileResponse.userToken) &&
      Objects.equals(this.deletedUntil, deleteUserProfileResponse.deletedUntil)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(userToken, deletedUntil);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DeleteUserProfileResponse {\n");
    sb
      .append("    userToken: ")
      .append(toIndentedString(userToken))
      .append("\n");
    sb
      .append("    deletedUntil: ")
      .append(toIndentedString(deletedUntil))
      .append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
